iam-login-service/src/main/java/it/infn/mw/iam/api/client/service/DefaultClientService.java

```diff
@@ -65,6 +69,9 @@ public DefaultClientService(Clock clock, IamClientRepository clientRepo,
   public ClientDetailsEntity saveNewClient(ClientDetailsEntity client) {
     client.setCreatedAt(Date.from(clock.instant()));
     eventPublisher.publishEvent(new ClientCreatedEvent(this, client));
+    if (!isNull(client.getClientSecret())) {
```

Comment:
In teoria abbiamo visto che non dovrebbe essere nullo ma stringa vuota.
Se è così, `IamClientSecretEncoder` è abile nel trattare stringhe vuole.
Possiamo verificare? Se è così questo check potrebbe non servire